scraper/keldoc/keldoc_center_scrap.py

```diff
@@ -0,0 +1,178 @@
+import csv
+import json
+import multiprocessing
+import os
+from typing import List, Optional
+
+import httpx
+
+from scraper.doctolib.doctolib_center_scrap import doctolib_urlify
+from utils.vmd_config import get_conf_platform, get_conf_inputs
+from utils.vmd_logger import get_logger
+
+KELDOC_CONF = get_conf_platform("keldoc")
+KELDOC_API = KELDOC_CONF.get("api", {})
+SCRAPER_CONF = KELDOC_CONF.get("center_scraper", {})
+BASE_URL = KELDOC_API.get("scraper")
+
+timeout = httpx.Timeout(KELDOC_CONF.get("timeout", 25), connect=KELDOC_CONF.get("timeout", 25))
+KELDOC_ENABLED = KELDOC_CONF.get("enabled", False)
+KELDOC_HEADERS = {
+    "User-Agent": os.environ.get("KELDOC_API_KEY", ""),
+}
+session = httpx.Client(timeout=timeout, headers=KELDOC_HEADERS)
+logger = get_logger()
+
+
+def run_departement_scrap(departement: str) -> list:
+    logger.info(f"[Keldoc centers] Parsing pages of departement {departement} through department SEO link")
+    centers_departements = parse_pages_departement(departement)
+    if centers_departements == 0:
+        raise Exception("No Value found for department {}, crashing")
+    return centers_departements
+
+
+def send_keldoc_request(url: str) -> Optional[dict]:
+    try:
+        req = session.get(url)
+        req.raise_for_status()
+        return req.json()
+    except httpx.TimeoutException as hex:
+        logger.warning(f"Keldoc request timed out: {url}")
+        return None
+    except httpx.HTTPStatusError as hex:
+        logger.warning(f"Keldoc request returned HTTP code {hex.response.status_code}: {url}")
+        return None
+    except (httpx.RemoteProtocolError, httpx.ConnectError) as hex:
+        logger.warning(f"Keldoc raise error {hex} for request: {url}")
+        return None
+
+
+def parse_keldoc_centers(page_limit=None) -> List[dict]:
+    centers = []
+    unique_center_urls = []
+
+    with multiprocessing.Pool(50) as pool:
+        center_lists = pool.imap_unordered(run_departement_scrap, get_departements())
+        centers = []
+
+        for center_list in center_lists:
+            centers.extend(center_list)
+
+        for item in list(centers):
+            if item.get("rdv_site_web") in unique_center_urls:
+                centers.remove(item)
+                continue
+            unique_center_urls.append(item.get("rdv_site_web"))
+
+        return centers
+
+
+def get_departements():
+    NOT_INCLUDED_DEPARTEMENTS = []
+    with open(get_conf_inputs().get("departements"), encoding="utf8", newline="\n") as csvfile:
+        reader = csv.DictReader(csvfile)
+        departements = [str(row["nom_departement"]) for row in reader]
+        [departements.remove(ndep) for ndep in NOT_INCLUDED_DEPARTEMENTS]
+        return departements
+
+
+def parse_keldoc_resources(center: dict) -> dict:
+    center_url = center.get("url")
```

Comment:
Tu `split` après, donc ça fail de toute manière s’il n’y a pas de match.

```suggestion
    center_url = center["url"]
```